pkg/server/config/config.go

```diff
@@ -61,6 +61,7 @@ type Database struct {
 type AuthenticatingProxy struct {
 	Hostname       string `yaml:"hostname"`
 	UsernameHeader string `yaml:"username_header"`
+	SpiffeId       string `yaml:"spiffe_id"`
```

Comment:
This is not something that we do today, but it would be great if we validated config on start-up. E.g. we should check that SPIFFE IDs are valid.

Separately from this, I wonder if we wanted to make this an array, something like `AllowedSPIFFEIDs` since multiple authenticating proxies could be trying to connect to sharkey.